.github/workflows/build-and-test.yml

```diff
@@ -6,16 +6,22 @@ permissions:
   contents: read
 
 jobs:
-  check:
+  lint:
     name: Check Code with stable output
     runs-on: ubuntu-latest
     steps:
       - name: Install protoc
         run: sudo apt install -y protobuf-compiler
-      - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@stable
-      - name: run check
-        run: make check
```

Comment:
We do have a clippy run in [check](https://github.com/bitcoin-dev-project/sim-ln/blob/main/Makefile#L48), perhaps we just need to add `--all-targets --color always -- --deny warnings`? 

Noticed that all the fixes are in tests, so that might be what we're missing?

Reply:
Should be fixed in [877ac51](https://github.com/bitcoin-dev-project/sim-ln/pull/177/commits/877ac51d571275c79978f6d7b5412f53b18f7dde)